.travis.yml

```diff
@@ -6,26 +6,30 @@ branches:
     - /^\d\.\d+\.\d+(rc\d+|\.dev\d+)?$/
 matrix:
   include:
-    - python: 2.7
-      env: TOXENV=py27
-    - python: 2.7
-      env: TOXENV=jessie
-    - python: 2.7
-      env: TOXENV=pypy
-    - python: 2.7
-      env: TOXENV=pypy3
-    - python: 3.4
-      env: TOXENV=py34
-    - python: 3.5
-      env: TOXENV=py35
-    - python: 3.6
-      env: TOXENV=py36
-    - python: 3.7
-      env: TOXENV=py37
+    - env: TOXENV=py27
       dist: xenial
-      sudo: true
-    - python: 3.6
-      env: TOXENV=docs
+      python: 2.7
+    - env: TOXENV=pypy
+      dist: xenial
+      python: 2.7
+    - env: TOXENV=stretch
```

Comment:
Sorry to pop this topic "again" 🙏 . 

I don't see the point of pinning specific python packages for *Debian* (Jessie in the past, now Stretch) but testing it under an Ubuntu distribution.

IMO we should aim to define the baseline of system libraries (i.e: libssl, libxml2(?)...) which are required to install Scrapy and its dependencies. Furthermore, considering Travis-CI is our CI service, its build environment is Ubuntu based and that Ubuntu LTS releases had proven to be reliable; I vote to choose from Trusty, Xenial or Bionic as system baseline. I vogue to drop the idea of Debian as baseline simply because we can't test on it (at least for now). 

Then we intersect what the baseline system can make platform dependent packages like _pyOpenSSL_ or _cryptography_ works with, with what major libraries Scrapy depends-on (i.e. Twisted). The minimum system libraries available by the Linux distribution of choice in addition to the minimum python package versions required by Scrapy to work are what define the baseline. 

Nowadays the only things that can't be overridden in a Python virtual environment are system libraries, while Python packages can be easily isolated from the host system. IMHO Python packages packaged by distributions can't be considered constrains for Scrapy baseline. For example, if we choose Xenial as baseline system, it doesn't matter what Twisted version is provided by `python-twisted` deb package as long as Twisted can be installed without affecting the host system.

In that sense, and if all maintainers agree, I'd remove this entry from the matrix and all related entries in `tox.ini`

Reply:
> Python packages packaged by distributions can't be considered constrains for Scrapy baseline

I agree with this.

Reply:
Since [Travis CI supports Docker](https://docs.travis-ci.com/user/docker/), I wouldn’t discard Debian-based tests just because Travis CI Linux images are all Ubuntu.

That said, I don’t have an opinion on Debian vs Ubuntu (vs any other Linux distribution).

Reply:
Agreed, supporting working on system Python modules is too hard and doesn't seem useful to me.

Reply:
1) I think that's fine to not use Debian as a baseline in any way, it'd be simpler for us.
2) There is still value in allowing Scrapy to run with not the latest lxml / Twisted / cryptography(?) / Pillow, as there can be other software running in the same environment as Scrapy, which is not upgraded to the latest version yet; also, there can be regressions. We may decide to support only recent enough versions not tied to any distribution, but it should be clear what these versions are, and we should test on these versions explicitly, to make sure Scrapy works.

Reply:
So we should test on the minimal supported versions (which we should choose and put in the docs), on "current" versions and maybe on some versions in between.

Reply:
Yes; having versions "in between" can be an overkill though, unless we want to support a really wide range of versions for some package.

Reply:
> ... We may decide to support only recent enough versions not tied to any distribution, but it should be clear what these versions are, and we should test on these versions explicitly, to make sure Scrapy works.

It makes sense, let's add a "baseline" tox env for 1.8 that is an upgrade of current Jessie env but with versions that works on Xenial, and once we start working on 2.0 we bump it again taking Bionic as system base.